### DISCUSSION TITLE
Ideas for optimising offline sessions

### DISCUSSION THREAD

**Discussion by thomasdarimont** [keycloak/keycloak#24263] [Ideas]
This discussion should serve as a place for collecting ideas for optmizing the offline session feature.

If you have an Idea just create dicsussion and add a link in the comments, well collect the curated ideas here.

# Ideas
- [Use shorter lifetimes for offline sessions cache entries in memory](https://github.com/keycloak/keycloak/discussions/24274)

**Comment by stianst** [comment]
Would be good to also consider non-offline sessions if we do anything more fancy around offline sessions. Frequently offline sessions are used as a work-around to the fact that regular sessions are not durable, which is not the right use-case for them. Offline sessions are intended for the use-case of giving permanent access to an application on a users behalf, not for example for durable/long-term session for a mobile app.

Regular sessions could have a very long expiration, and also consume a lot of memory (always, even though they can be infrequently accessed). For this reason it would be good to remove the difference between regular and offline sessions in how they are stored, or at least supporting storing regular sessions in a similar way to offline sessions.

**Reply by dasniko** [reply]
> For this reason it would be good to remove the difference between regular and offline sessions in how they are stored, or at least supporting storing regular sessions in a similar way to offline sessions.

I'd love to see something like this! 🚀 💯 👍
